README.md

```diff
@@ -215,17 +216,87 @@ Besides this `README.md` document, the project includes extensive documentation:
 - [`doc/c64-openapi.yaml`](doc/c64-openapi.yaml) — OpenAPI 3.1 description of the REST surface.
 
-## Configuration ⚙️
+## Configuration
+
+The MCP server reads configuration from a JSON file called `.c64mcp.json`. The recommended location is your home directory (`~/.c64mcp.json`). You can override the path with the `C64MCP_CONFIG` environment variable. As a convenience during development, a project-local [`.c64mcp.json`](.c64mcp.json) at the repo root is also picked up if present.
+
+From this release onward, configuration is split by device type. No top-level `backend` field is required; the server selects a backend automatically (see selection rules below).
+
+### C64U (real hardware)
+
+Use this section to point the server at an Ultimate 64/Commodore 64 Ultimate device. Either `hostname` or an explicit `baseUrl` can be provided.
+
+```json
+{
+  "c64u": {
+    "hostname": "c64u",
+    "baseUrl": "http://192.168.1.13"
+  }
+}
+```
+
+Notes:
+
+- If both `hostname` and `baseUrl` are set, `baseUrl` wins.
+- The legacy keys `c64_host` and `baseUrl` are still recognized for backward compatibility; they map to the same effective target.
+
+### VICE (emulator, experimental)
+
+This backend starts a fresh VICE process for each PRG run using the emulator binary. In phase one, memory/register operations are not supported; the focus is deterministic PRG execution.
+
+```json
+{
+  "vice": {
+    "exe": "/usr/bin/x64sc"
+  }
+}
+```
+
+Notes:
+
+- If `vice.exe` is not set, the server attempts to find `x64sc` (or `x64`) on your `PATH`.
+- Each program execution spawns a new VICE instance, e.g.:
+
+  ```bash
+  x64sc -autostart "program.prg" -silent -warp
+  ```
+
+### Backend selection rules
+
+Backend selection is automatic with clear logging. The following precedence applies:
+
+1. Explicit override: if `C64_MODE=c64u` or `C64_MODE=vice` is set in the environment, that backend is used.
+2. Config presence: if only one of `c64u` or `vice` is configured, it is used.
+3. Both configured: prefer `c64u` unless VICE is explicitly requested via `C64_MODE=vice`.
+4. No configuration: probe the default C64U address (`http://c64u`); if unavailable, fall back to VICE.
 
-  The `c64_host` value can be either a hostname (e.g. `c64u`) or an IP address. Save the file as `~/.c64mcp.json`.
+On startup, the server logs the selected backend and reason, for example:
 
-  You can override the path with the `C64MCP_CONFIG` environment variable.
+- `Active backend: c64u (from config)`
+- `Active backend: vice (fallback – hardware unavailable)`
 
-   If the file is missing, the server first looks for the bundled [`.c64mcp.json`](.c64mcp.json) in the project root, and finally falls back to `http://c64u`.
+### File locations and overrides
 
+- Primary: `~/.c64mcp.json`
+- Override path: set `C64MCP_CONFIG=/absolute/path/to/.c64mcp.json`
+- Repo-local (dev): `.c64mcp.json` at the project root
 
+### Backward compatibility
```

Comment:
@cursoragent This is not needed - remove the backward compatibility in the code, as well as this chapter. 

Then generalize the tests that interact with the C64U to instead refer to the C64Facade. 

Finally, extend the tests of the C64Facade to run twice where applicable: once against the C64U implementation of the facade, and once against the Vice implementation, keeping in mind the following:
- If a device (c64u, device) does not support a certain tooling endpoint (as per the tool method annotations), then the test expects an error response.
- Otherwise, it expects a valid response.

Technical Notes:
- Start vice headless using xvfb, otherwise it is bound to fail on CI.